src/api.js

```diff
@@ -752,28 +777,26 @@ api.declare({
     return res.reportError(detail.code, detail.message, detail.details);
   }
 
-  // Extra scopes required
-  let scopes = _.flatten([
-    // task.scopes
-    taskDef.scopes,
-
-    // Find scopes required for task specific routes
-    taskDef.routes.map(route => 'queue:route:' + route),
-
-    // Add scope for priority if any
-    taskDef.priority !== 'normal' ? [
-      'queue:task-priority:' + taskDef.priority,
-    ] : [],
-  ]);
-
-  // Authenticate request by providing parameters, and then validate that the
-  // requester satisfies all the scopes assigned to the task
-  if (!req.satisfies({
-    provisionerId:  taskDef.provisionerId,
-    workerType:     taskDef.workerType,
-    schedulerId:    taskDef.schedulerId,
-    taskGroupId:    taskDef.taskGroupId,
-  }) || !req.satisfies([scopes])) {
+  // Authenticate with legacy scopes first
+  let {provisionerId, workerType, schedulerId, taskGroupId} = taskDef;
+  let legacyAuthorized = req.satisfies([[
+    `queue:define-task:${provisionerId}/${workerType}`,
+  ], [
+    `queue:create-task:${provisionerId}/${workerType}`,
+  ], [
+    `queue:define-task:${provisionerId}/${workerType}`,
+    `queue:task-group-id:${schedulerId}/${taskGroupId}`,
+  ]], true);
+
+  // Require 'queue:create-task:<priority>:<provisionerId>/<workerType>' for
+  // given priority or higher...
+  let priorities = PRIORITY_LEVELS.slice(0, PRIORITY_LEVELS.indexOf(taskDef.priority) + 1);
+  assert(priorities.length > 0, 'must have a non-empty list of priorities');
+  // If not authorized by legacy, we require most recent scopes
+  if (!legacyAuthorized && !req.satisfies(priorities.map(priority => [
+    `queue:create-task:${priority}:${provisionerId}/${workerType}`,
+    `queue:scheduler-id:${schedulerId}`,
+  ]))) {
```

Comment:
We're missing the `taskDef.scopes` and `taskDef.routes` checks in `defineTask` now, right?